### PR TITLE
Fixes Geometry.segment_intersects_circle working only one way.

### PR DESCRIPTION
**Pull request by Schroedi**
Fixes line segment circle intersection tests for most cases. For details see: #26698

### CODE REVIEW

File: core/math/geometry.h [hunk 1]
```diff
@@ -702,9 +702,11 @@ class Geometry {
 		/* if we can assume that the line segment starts outside the circle (e.g. for continuous time collision detection) then the following can be skipped and we can just return the equivalent of res1 */
 		sqrtterm = Math::sqrt(sqrtterm);
 		real_t res1 = (-b - sqrtterm) / (2 * a);
-		//real_t res2 = ( -b + sqrtterm ) / (2 * a);
+		real_t res2 = (-b + sqrtterm) / (2 * a);
 
-		return (res1 >= 0 && res1 <= 1) ? res1 : -1;
+		if (res1 >= 0 && res1 <= 1) return res1;
+        if (res2 >= 0 && res2 <= 1) return res2;
```
Comment (akien-mga) [inline core/math/geometry.h]: Indentation issue.

Comment (Schroedi) [reply core/math/geometry.h]: Just noticed :D

**Comment by reduz** [comment]
seems good

**Comment by akien-mga** [comment]
Thanks!

**Comment by hpvb** [comment]
Consider for 3.1.2

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**